projects/core/dsl/runtime.py

```
@@ -118,21 +130,25 @@ def execute_tasks(function_args: dict = None):
-                if execution_error is None:
-                    raise
-                # If both failed, log always task error but preserve original error
-                logger.error(f"==> ALWAYS TASK ALSO FAILED: {e}")
-                logger.info("")
+            # Always execute "always" tasks from this file
+            if file_tasks:
+                logging.warning("Executing the @always tasks ...")
+                while file_tasks:
+                    try:
+                        current_task_info = file_tasks.pop(0)
+                        _execute_single_task(task_info, args, shared_context)
```

Ruff (0.15.10)

[error] Undefined name `task_info`

(F821)